src/Compilers/CSharp/Portable/Errors/CustomObsoleteDiagnosticInfo.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Errors
+{
+    internal sealed class CustomObsoleteDiagnosticInfo : DiagnosticInfo
```

Comment:
>DiagnosticInfo [](start = 57, length = 14)

Should we inherit from CSDiagnosticInfo?

Reply:
Possibly, what's the benefit of inheriting from CSDiagnosticInfo instead of DiagnosticInfo?

Reply:
> Possibly, what's the benefit of inheriting from CSDiagnosticInfo instead of DiagnosticInfo?

I think that is something that you should figure out, look at other examples of subclasses of DiagnosticInfo within the compiler, etc. 

---
In reply to: [396841277](https://github.com/dotnet/roslyn/pull/42518#discussion_r396841277) [](ancestors = 396841277)

Reply:
It turned out to be convenient to be able to share this diagnostic info between C# and VB, and looking at what the CSDiagnosticInfo adds it doesn't seem to help to inherit from it.